### PR TITLE
Configurable storage volumes

### PR DESCRIPTION
**Pull request by ancorgs**
## Problem

The `Y2Storage::GuidedSetup` tries to accommodate a set of volumes. That's configurable in YaST (and is usually defined by the product, the installation role and other criteria). But in D-Installer there was no way to specify the set of volumes, so it always created the fallback set of volumes that is hard-coded at YaST.

## Solution

Now `d-installer.yaml` can contain a 'volumes' section that is identical to the [corresponding section](https://github.com/yast/yast-installation/blob/master/doc/control-file.md#the-volumes-subsection) of YaST configuration (just using Yaml instead of XML, of course).

The code still uses the fallback legacy list if the list of volumes is missing in the configuration file, but this pull request also syncs the current `d-installer.yaml` with the Tumbleweed configuration [available at openSUSE's skelcd](https://github.com/yast/skelcd-control-openSUSE/blob/master/control/control.openSUSE.xml#L249) repo.

## Testing

- Tested manually with:
  - The hard-coded fallback
  - The TW-like configuration mentioned above
  - A configuration equivalent to MicroOS
- Included the first unit tests for `DInstaller::Storage::Proposal`

### CODE REVIEW

File: service/etc/d-installer.yaml [hunk 1]
```diff
@@ -12,3 +12,72 @@ web:
...
+      archs: riscv64
+
+  - mount_point: "/home"
+    fs_type: xfs
+    desired_size: 40 GiB
+    min_size: 10 GiB
+    max_size: unlimited
+    weight: 60
+
+    proposed: false
```
Comment (jreidinger) [inline service/etc/d-installer.yaml:68]: I think we should at least link current existing documentation, as some keys looks strange for me.

File: service/lib/dinstaller/storage/proposal.rb [hunk 2]
```diff
@@ -120,11 +125,22 @@ def calculate(settings = {})
       def generate_proposal_settings(settings)
         proposal_settings = Y2Storage::ProposalSettings.new_for_current_product
 
+        config_volumes = read_config_volumes
+        proposal_settings.volumes = config_volumes unless config_volumes.empty?
```
Comment (jreidinger) [inline service/lib/dinstaller/storage/proposal.rb:129]: what means here empty config volumes? That product does not require any partitioning?

File: service/etc/d-installer.yaml [hunk 3]
```diff
@@ -12,3 +12,72 @@ web:
...
+    # There must always be a root
+    proposed_configurable: false
+
+    snapshots: true
+    snapshots_percentage: 250
+    snapshots_configurable: true
+    # Disable snapshots if there is not enough room
+    disable_order: 3
+
+    btrfs_default_subvolume: "@"
```
[the remark this answers is not in the document]
Comment (ancorgs) [reply service/etc/d-installer.yaml:34]: Not all. Default values and other aspects are explained in the documentation for the YaST control file, that is now linked.

File: service/etc/d-installer.yaml (hunk 1 above)
Comment (ancorgs) [reply service/etc/d-installer.yaml:68]: Of course, the plan was to link to the corresponding YaST documentation. Now it's done.

File: service/lib/dinstaller/storage/proposal.rb (hunk 2 above)
Comment (ancorgs) [reply service/lib/dinstaller/storage/proposal.rb:130]: I added a comment (and a test case) to clarify that.

**Review by jreidinger — APPROVED** [review]

**Comment by ancorgs** [comment]
@jreidinger This would be the corresponding section for microos
https://paste.opensuse.org/49605740

**Pull request merged**